src/plugins/intel_cpu/thirdparty/CMakeLists.txt

```diff
@@ -169,4 +169,13 @@ if(ENABLE_SHL_FOR_CPU)
     ov_install_static_lib(shl ${OV_CPACK_COMP_CORE})
 endif()
 
+if(ENABLE_KLEIDIAI_FOR_CPU)
+    set(KLEIDIAI_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+    set(TEMP_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+    set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libs" FORCE)
```

Comment:
don't override `BUILD_SHARED_LIBS` via cache, you can set it locally similar to 
https://github.com/openvinotoolkit/openvino/blob/b1358adcd4d4964db8f6ffc86b4a64e9e695bfa5/thirdparty/dependencies.cmake#L280-L284
via cmake `function`, which limits scope of variables.